add  config.reopen_browser_on_fail

(currently it's kind of "always true" with not ability to set it to false)

Tested on selene 2.0.0a35+ current master https://github.com/yashaka/selene/tree/0e7ed43957d08ff37f37c731e8fc9136500b85c4
```python
import pytest
from selenium import webdriver
from webdriver_manager.chrome import ChromeDriverManager

from selene import by, be, Browser, Config
from selene.support.shared import browser


@pytest.fixture(scope='session')
def session_driver():
    chrome_driver = webdriver.Chrome(ChromeDriverManager().install())
    yield chrome_driver
    chrome_driver.quit()


@pytest.fixture(scope='function')
def session_browser(session_driver):
    yield Browser(Config(driver=session_driver))


def test_reopen(session_browser):
    session_browser.open('http://google.ru')
    session_browser.element(by.name('q')).should(be.visible)
    session_browser.quit()
    session_browser.element(by.name('q')).should(be.visible)  # <-- fails here


def test_reopen_shared():
    browser.open('http://google.ru')
    browser.element(by.name('q')).should(be.visible)
    browser.quit()
    browser.element(by.name('q')).should(be.visible)  # <-- fails here
```

`test_reopen` fails with
```
>                       raise self._hook_failure(failure)
E                       selene.core.exceptions.TimeoutException: Message: 
E                       
E                       Timed out after 4s, while waiting for:
E                       browser.element(('name', 'q')).is visible
E                       
E                       Reason: MaxRetryError: HTTPConnectionPool(host='127.0.0.1', port=52878): Max retries exceeded with url: /session/71916a02efa4ce65d0878f3f961084ea/element (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x11040ab00>: Failed to establish a new connection: [Errno 61] Connection refused'))
```
`test_reopen_shared` fails with
```
    @property
    def instance(self) -> WebDriver:
        if self._closed:
            raise RuntimeError(
>               'Webdriver has been closed. '
                'You need to call open(url) '
                'to open a browser again.'
            )
E           RuntimeError: Webdriver has been closed. You need to call open(url) to open a browser again.
```

--- a/tests/acceptance/driver_management_test.py
+++ b/tests/acceptance/driver_management_test.py
@@ -173,6 +173,21 @@ def test_built_driver_will_not_be_killed_if_post_configured_for_hold_driver(
     assert driver.title == 'Selene Test Page'  # ALIVE
 
 
+def test_built_driver_will_not_be_killed_if_post_configured_for_deprecated_hold_browser_open(
+    with_process_exit_teardown,
+):
+    browser = selene.Browser(selene.Config())
+    browser.open(empty_page)
+
+    with pytest.warns(DeprecationWarning):
+        browser.config.hold_browser_open = True
+
+    atexit._run_exitfuncs()
+
+    driver = persistent.Field.value_from(browser.config, 'driver')
+    assert driver.title == 'Selene Test Page'  # ALIVE
+
+
 def test_driver_built_on_copy__will_not_be_killed__if_copied_with_hold_driver(
     with_process_exit_teardown,
 ):
@@ -357,8 +372,6 @@ def test_browser_resurrects_itself_and_copy__after_copy_programmed_to_death_fore
     browser.quit()
     copy = browser.with_(rebuild_not_alive_driver=False)  # <- GIVEN
 
-    # driver = browser.driver  # this will not work (TODO: add test for it?)
-    #                          # but will work if rebuild_... set to True on browser
     browser.open(empty_page)  # <- WHEN
     driver = persistent.Field.value_from(browser.config, 'driver')
 
@@ -367,6 +380,25 @@ def test_browser_resurrects_itself_and_copy__after_copy_programmed_to_death_fore
     assert driver is browser.driver
     assert driver is copy.driver
     assert driver is persistent.Field.value_from(copy.config, 'driver')
+
+
+def test_browser_driver_access_rebuilds_or_not_depending_on_rebuild_not_alive_driver(
+    with_process_exit_teardown,
+):
+    browser = selene.Browser(selene.Config())
+    browser.quit()
+
+    dead_driver = browser.driver
+    assert dead_driver.name == 'chrome'
+    pytest.raises(MaxRetryError, lambda: dead_driver.title)
+    dead_session_id = dead_driver.session_id
+
+    browser.config.rebuild_not_alive_driver = True
+    rebuilt = browser.driver
+    assert rebuilt.title == ''  # alive
+    assert rebuilt is not dead_driver
+    assert rebuilt.session_id != dead_session_id
+    assert rebuilt is persistent.Field.value_from(browser.config, 'driver')
 
 
 def test_can_build_second_driver_if_previous_was_forgotten_via_reset_to_ellipsis(

